swarm/src/behaviour.rs

```diff
@@ -93,19 +93,22 @@ pub trait NetworkBehaviour: Send + 'static {
     /// The default implementation of this method calls `inject_disconnected` followed with
     /// `inject_connected`. This is a logically safe way to implement this behaviour. However, you
     /// may want to overwrite this method in the situations where this isn't appropriate.
-    fn inject_replaced(&mut self, peer_id: PeerId, closed_endpoint: ConnectedPoint, new_endpoint: ConnectedPoint) {
-        self.inject_disconnected(&peer_id, closed_endpoint);
-        self.inject_connected(peer_id, new_endpoint);
+    #[deprecated(since = "0.16.0", note =
+        "Connections are no longer replaced and this function no longer called. \
+        It will eventually be removed from the trait."
+    )]
+    fn inject_replaced(&mut self, _: PeerId, _: ConnectedPoint, _: ConnectedPoint) {
```

Comment:
Considering that we're renaming `inject_node_event` to `inject_event`, I'd remove that method immediately and therefore remove confusion.

Reply:
Alright.